partcad/src/partcad/lint/schema.py

```
@@ -59,3 +65,50 @@ async def validate(self, ctx: Context, package: Project, target: str, lint_ctx:
+        lint_result = LintingReport(package.name)
+
+        async with aiofiles.open(target, mode="r") as file:
+            try:
+                raw = await file.read()
+            except (OSError, IOError, UnicodeDecodeError) as err:
+                lint_result.add(Severity.FAILED, f"Failed to read assembly file: {err}")
+                return lint_result
+
+        try:
+            diagnostics = validate_source(raw, get_schema(ASSY_SCHEMA))
+        except Exception as exc:  # pylint: disable=broad-except
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)